unidiff/patch.py

```diff
@@ -473,8 +475,10 @@ def _parse(self, diff, encoding, metadata_only):
             # check for a git file rename
             is_diff_git_header = RE_DIFF_GIT_HEADER.match(line)
             if is_diff_git_header:
-                if patch_info is None:
-                    patch_info = PatchInfo()
+                if current_file is not None and not is_file_appended:
```

Comment:
I think we could refactor this to avoid introducing the `is_file_appended` flag by always setting up a `PatchedFile` when a git header is found (and some minor changes to accommodate for that), will check.

Reply:
AFAI remember I was in a rush of fixing it. Most probably, there is a more elegant way to fix it. I am 100% sure you are more confident with code! =)